highfidelity/hifi
# Add Equip Features to Edit.js

Blocked by this bug: https://highfidelity.fogbugz.com/f/cases/1450/in-vive-hand-controller-clicks-fall-through-html5-overlays-like-edit-js-and-hit-what-s-behind-it

This PR makes it so that you can edit the left & right hand position & rotation of equippable objects in edit.js.  It adds an Equippable toggle and shows/hides fields for the position & rotation for each hand.

To test:
1. Run default scripts from this repository
2. Find an equippable object (ping pong gun, bow and arrow, etc).
3. Open edit and select that item
4. See the new 'Equippable' toggle and section.  If the item is already equippable, that section should be prefilled
5. Make adjustments to the position and rotation of the equip info using the arrows.  Make adjustments using the number input
6. Observe that if you choose a position that would auto-unequip the item, you get an alert and the values revert to their previous values
7. Equip the object and observe that it is in a different place.
8. With an item equipped, make adjustments to the position + rotation.  You should see the object adjust in your hand in realtime.
9. With an item equipped, uncheck 'Equippable'.  The item should unequip.  

https://highfidelity.fogbugz.com/f/cases/1353/Equip-button-in-JS-with-positional-data-field
